dialect.go

```diff
@@ -310,13 +310,22 @@ func (m ClickHouseDialect) dbVersionQuery(db *sql.DB) (*sql.Rows, error) {
 }
 
 func (m ClickHouseDialect) insertVersionSQL() string {
-	return fmt.Sprintf("INSERT INTO %s (version_id, is_applied) VALUES ($1, $2)", TableName())
+	// TODO(mf): how did this work before?
+	// Note, throughout the codebase is_applied is always true on inserts, so maybe hard-coding this
+	// is fine?
+	// There is no separate type for boolean values. Use UInt8 type, restricted to the values 0 or 1.
+	// Ref: https://clickhouse.com/docs/en/sql-reference/data-types/boolean/
+	return fmt.Sprintf("INSERT INTO %s (version_id, is_applied) VALUES ($1, 1)", TableName())
 }
 
 func (m ClickHouseDialect) migrationSQL() string {
 	return fmt.Sprintf("SELECT tstamp, is_applied FROM %s WHERE version_id = $1 ORDER BY tstamp DESC LIMIT 1", TableName())
 }
 
 func (m ClickHouseDialect) deleteVersionSQL() string {
-	return fmt.Sprintf("ALTER TABLE %s DELETE WHERE version_id = $1", TableName())
+	// TODO(mf): deletes are asynchronous, but need to rethink whether this makes sense here?
+	// Ref: https://clickhouse.com/docs/en/operations/settings/#mutations_sync
+	// Ref: https://clickhouse.com/docs/en/sql-reference/statements/alter/delete/
+	// Ref: https://clickhouse.com/docs/en/faq/operations/delete-old-data/
+	return fmt.Sprintf("ALTER TABLE %s DELETE WHERE version_id = $1 SETTINGS mutations_sync = 2", TableName())
```

Comment:
I did this to make the tests pass, but I don't think we should be adding a mutations_sync setting. Instead in our tests we should assume `mutations_sync = 0` (async) and retry with a threshold.

E.g., a down migration deletes the migration entry from the goose table, and we would like to test for this behaviour that _eventually_ that row got deleted.

cc @huynguyenh @kshvakov @qonsth @kafkiansky @rickydjohn

(sorry for the ping, looking for some advice on ClickHouse if ya'll have a spare second)

cc @huynguyenh how are down migrations usually handled in ClickHouse, is it acceptable that a tool like goose triggers a delete and it happens async? 

What about when there are dependant migrations, are these split up and applied at different times? Or is there a setting that's usually set outside of goose to make it synchronous?